AdyenCashAppPay/CashAppPayComponent.swift

```diff
@@ -225,24 +231,37 @@ extension CashAppPayComponent: CashAppPayObserver {
         case let .approved(request, grants):
             submitApprovedRequest(with: grants, profile: request.customerProfile)
         case let .apiError(error):
-            fail(with: error)
+            fail(with: error, message: ErrorMessage.apiError)
         case let .networkError(error):
-            fail(with: error)
+            fail(with: error, message: error.localizedDescription)
         case let .unexpectedError(error):
-            fail(with: error)
+            fail(with: error, message: ErrorMessage.unexpectedError)
         case let .integrationError(error):
-            fail(with: error)
+            fail(with: error, message: ErrorMessage.integrationError)
         case .declined:
-            fail(with: Error.declined)
+            let error = Error.declined
+            fail(with: error, message: error.localizedDescription)
         default:
             break
         }
     }
 
-    private func fail(with error: Swift.Error) {
+    private func fail(with error: Swift.Error, message: String? = nil) {
         stopLoading()
+        sendThirdPartyErrorEvent(with: message)
         delegate?.didFail(with: error, from: self)
     }
+    
+    private func sendThirdPartyErrorEvent(with message: String?) {
```

Comment:
So this method repeats again. What do you think of having an additional generic method in `analyticsProvider`, something like 
```swift
analyticsProvider.addError(
    paymentMethod: paymentMethod.type.rawValue,
    errorType: .thirdParty,
    message: message)
```

Reply:
I am not a fan of repeating this either but maybe we can replace the `add(error)` function completely then as having 2 different ones also doesn't seem useful.

Reply:
Agreed. You have this all loaded so I'm just trying to provide feedback but please make your own decision on how to simplify this.